src/ilasm/asmman.cpp

```diff
@@ -121,8 +121,9 @@ AsmManAssembly*     AsmMan::GetAsmRefByName(__in __nullterminated const char* sz
         //ret = m_AsmRefLst.FIND(&X);
         //X.szAlias = NULL;
         DWORD L = (DWORD)strlen(szAsmRefName);
-        for(int i=0; (ret = m_AsmRefLst.PEEK(i))&&
-            ((ret->dwAlias != L)||strcmp(ret->szAlias,szAsmRefName)); i++);
+        for (int i = 0; (ret = m_AsmRefLst.PEEK(i)); i++) {
+            if (((ret->dwAlias == L) && !strcmp(ret->szAlias, szAsmRefName))) break;
```

Comment:
Looks like you have two pairs of extra brackets.

for (int i = 0; ret = m_AsmRefLst.PEEK(i); i++) {
if ((ret->dwAlias == L) && strcmp(ret->szAlias, szAsmRefName) == 0) break;
}